### PR TITLE
Replace distutils with packaging

### LINKED ISSUE #2121: DeprecationWarning: The distutils package is deprecated
## 🐛 Bug

TorchMetrics still uses distutils, but distutils is deprecated.

### To Reproduce

```console
> python3 -Wall
>>> import torchmetrics
lib/python3.11/site-packages/torchmetrics/utilities/imports.py:18: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.version import LooseVersion
```

### Expected behavior

I would not expect TorchMetrics to still be using distutils.

### Environment

- TorchMetrics version: 1.1, 1.2, master (spack)
- Python & PyTorch Version: 3.11.4 & 2.0.1
- Any other relevant information such as OS: macOS 13.5.2

### Additional context

I also sometimes see a different error message from pytest:
```
DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
```
This gives one possible solution.

### PR DESCRIPTION
**Pull request by adamjstewart**
## What does this PR do?

Fixes #2121

This will be required for Python 3.12 support

@Borda 

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--2137.org.readthedocs.build/en/2137/

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #  in case you want to preserve/enforce restrictions on the latest compatible version, add "strict" as an in-line comment
 
 numpy >1.20.0
+packaging
```
Comment (Borda) [inline requirements/base.txt:5]: any min version?

**Review by Borda — APPROVED** [review]

File: requirements/base.txt (hunk 1 above)
Comment (adamjstewart) [reply requirements/base.txt:5]: Not sure, the API is pretty stable so I imagine these functions have existed for a long time. Do you test the minimum version in CI? I could try a binary search for a minimum version that passes the tests.

Comment (Borda) [reply requirements/base.txt:5]:
```suggestion
packaging >17.1
```
so, set it at least 5 years back?

Comment (adamjstewart) [reply requirements/base.txt]: Should we do the same for typing-extensions? I hate to choose an arbitrary version...

Comment (Borda) [reply requirements/base.txt]: let's pick one which was released 3-5 years ago :)

**Review by justusschock — APPROVED** [review]

**Pull request merged**